test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1240,16 +1241,29 @@ public synchronized void validateClusterFormed() {
         }
         logger.trace("validating cluster formed, expecting {}", expectedNodes);
 
-        assertFalse(
-            client().admin()
-                .cluster()
-                .prepareHealth(TEST_REQUEST_TIMEOUT)
-                .setWaitForEvents(Priority.LANGUID)
-                .setWaitForNodes(Integer.toString(expectedNodes.size()))
-                .get(TimeValue.timeValueSeconds(40))
-                .isTimedOut()
-        );
         try {
+            assertBusy(() -> {
+                try {
+                    final boolean timeout = client().admin()
```

Comment:
This call may fail when client user has role with application privileges and security index is still not available for searching. See my other https://github.com/elastic/elasticsearch/pull/120323#discussion_r1925285153